### ISSUE TITLE
Refine roc_sender/roc-receiver API

### ISSUE THREAD

**Issue by gavv** [roc-project/roc#263]
Currently we have roc_sender and roc_receiver API which has the following flow:

* roc_receiver binds to endpoints
* roc_sender connects to that endpoints
* the user writes the stream to roc_sender and reads from roc_receiver

The problem is that with this API, the sender is always a client (the one who connects) and the receiver is always a server (the one who binds). This is always the case for bare RTP/FECFRAME, but with RTSP, which we're currently implementing, the opposite case becomes also possible:

* roc_sender binds to endpoints
* roc_receiver connects to that endpoints
* the user writes the stream to roc_sender and reads from roc_receiver

So we need to improve the API in the following way:

* both roc_sender and roc_receiver should have both bind() and connect() methods
* the user should bind() one of them to endpoint(s) and connect another one the that endpoint(s), but it's up to the user which one to bind and which one to connect

In other words, both roc_sender and roc_receiver may act as server or as a client, depending on the user choice.

We should also update the Doxygen documentation (describe the new bind/connect scheme and endpoints).

**Comment by alexandremgo** [comment]
I can work on this issue if help is wanted ?

**Comment by gavv** [comment]
Hi Alexandre! Yeah, help is always wanted :)

This issue can not be resolved right now: we first need to implement session negotiation with RTSP (#216, #34), for which we first need to implement roc_ctl module (#214, #244, #215), SDP support (#200), and TCP support (#205). We also should finish migration from ports to URIs (#258).

This issue requires RTSP because currently we support only bare RTP/FECFRAME, which just transfers audio in one direction, from sender to receiver. Because of this, the sender has to be client (the one who connects) and the receiver has to be server (the one who binds).

If we want to use the sender as a server and the receiver as a client, receiver should connect to sender and tell it to start RTP transmission to the receiver. RTSP is a signaling protocol that will allow us to do it.

All these changes are planned to upcoming 0.2 release. Currently @dshil works on TCP support and I've started working on the URI support (#264, #270, #281, #282) but then took a break to earn some money. I'm planning to freelance until New Year and then will switch back to 0.2 for about a month.

If you want to help with 0.2, you're welcome. We have to resolve issues from the bottom to the top in our case. So here are 0.2 issues that can be started right now and are not assigned to anyone: SDP support, using Ragel (#200), RTCP support, via hand-written parser/composer (#14), control commands support in roc_pipeline (#215).

After adding control commands support we will be able to start working on roc_ctl. After adding SDP and TCP support we will be able to start working on RTSP and integrating it into roc_ctl. RTCP support is quite standalone but it would be highly desirable to include it in 0.2 as well.
